packages/react/src/validationSchema.tsx

```diff
@@ -126,14 +124,47 @@ const validatorForField = (field: FieldMetadata) => {
   for (const validation of field.configuration.validations) {
     switch (validation?.__typename) {
       case "GadgetRangeFieldValidation": {
-        const rangeValidation = validation as GadgetRangeFieldValidation;
+        const { min, max } = validation as GadgetRangeFieldValidation;
         if (validator instanceof StringSchema || validator instanceof NumberSchema) {
-          if (rangeValidation.min) {
-            validator = validator.min(rangeValidation.min);
+          if (min) {
+            validator = validator.min(min);
           }
-          if (rangeValidation.max) {
-            validator = validator.max(rangeValidation.max);
+          if (max) {
+            validator = validator.max(max);
           }
+        } else if (field.fieldType === GadgetFieldType.File && validator instanceof MixedSchema) {
+          validator = validator.test("is-valid-file-size", function (value) {
+            const { path, createError } = this;
+
+            if (isAutoFileFieldValue(value) && value.$invalidFileSize) {
+              if (min) {
+                if (max) {
+                  return createError({
+                    path,
+                    message: `must be a file between ${filesize(min)} and ${filesize(max)} big, and this file's size is ${filesize(
+                      value.$invalidFileSize
+                    )}.`,
+                  });
+                } else {
+                  return createError({
+                    path,
+                    message: `must be a file larger than than ${filesize(min)}, and this file's size is ${filesize(
+                      value.$invalidFileSize
+                    )}.`,
+                  });
+                }
+              } else {
+                if (max) {
+                  return createError({
+                    path,
+                    message: `must be a file smaller than ${filesize(max)}, and this file's size is ${filesize(value.$invalidFileSize)}.`,
+                  });
+                }
+              }
+            }
```

Comment:
The message is the only ting that varies here and the `createError` stuff is repeated. Could you keep just the messages in conditional blocks, and use a single return statement?
```ts
if (min && max) message="...";
else if (min) message="...";
else if (max) message="...";

return createError({message, ...})
```